pages/post/[interviewId].js

```diff
@@ -0,0 +1,57 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { styled } from "styled-components";
+import { interviews } from "@/public/interviews";
+
+export default function InterviewArticle() {
+  const router = useRouter();
+  const interviewId = router.query.interviewId;
+  const article = interviews.find((interview) => interview.id == interviewId);
+  console.log(interviewId);
+  return (
+    <article>
+      <TitleContainer>{article?.title}</TitleContainer>
+      <ImageContainer>
+        <Image src={article?.image} alt={article?.title} />
+      </ImageContainer>
+      <TextContainer>{article?.text}</TextContainer>
+    </article>
```

Comment:
Instead of checking every time with the `?.` operator, you can intercept it before like:

```js
const article = interviews.find((interview) => interview.id == interviewId);

if (!article) {
  redirect('/');
}
```

You'll have to import the [redirect](https://nextjs.org/docs/app/api-reference/functions/redirect) function at the top of your file, like `import { redirect } from 'next/navigation';`.

Reply:
The redirect threw an error so I kept the one we already used:
`if (!article) {
    return null;
  }`